spec/rollbar/plugins/delayed_job_spec.rb

```diff
@@ -34,6 +34,19 @@ def do_job_please!(a, b)
     end
   end
 
+  context 'with failed deserialization' do
+    let(:expected_args) do
+      [/Delayed::DeserializationError/, {:use_exception_level_filters=>true}]
+    end
+
+    it 'sends the exception' do
+      expect(Rollbar).to receive(:scope).with(kind_of(Hash)).and_call_original
+      allow_any_instance_of(Delayed::Backend::Base).to receive(:payload_object).and_raise(Delayed::DeserializationError)
+      expect_any_instance_of(Rollbar::Notifier).to receive(:error).with(*expected_args)
+
+      FailingJob.new.delay.do_job_please!(:foo, :bar)
```

Comment:
If I follow this test correctly, this line will cause delayed job to eventually call `do_job_please!`, the backend will call `payload_object` during the call to `JobData.new(job).to_hash`, this is stubbed to raise a deserialization error, which is rescued and therefore `to_hash` is able to return something with `handler_parent['handler']` therefore set to `{}`. The way things are set up with the plugin means that delayed job will make a failure lifecycle call, which we hook into and call report which is where we satisfy the expectation of the notifier receiving a call to `error` with the deserialization error as argument. A failure to deserialize the payload causes delayed job to short circuit to failing the job and never retrying so we have to report it this way.